content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,669 @@
+```
+
+You can check inside your `Pulumi.<stackname>.yaml` for which key you're using to encrypt your secrets:
+
+```yaml
+secretsprovider: awskms://alias/stack-encryption-key?region=us-west-2
+encryptedkey: AQICAHjqW3rb5Hw5Vpxi0c1sayz52VXj7yn20WVwsVILJSBU8wFDjvGuox3wDCJX99TxZFzAAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMWNeFRZIg8kVXMxrUAgEQgDtz6zV0aqegeAbmaQUNllMp8PQJa1qbjBH813I/XH6LbfynxZO9NE3sYPG89G0u/ltYsADiUAFS0bnadQ==
+config:
+  aws:region: us-west-2
+```
+
+## Use your Encrypted Secret
+
+Now we've initialized our stack we can add a secret configuration value using the secret flag and see the encrypted config value in the stack configuration:
+
+```bash
+pulumi config set --secret <projectname>:supersecret correct-horse-battery-stable
+```
+
+## Verify the encryption
+
+To verify that the secret is indeed only accessible to the KMS key we created earlier, we can remove access to the key temporarily and try to perform a Pulumi operation. First, let's use the secret in our stack:
+
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+
+const config = new pulumi.Config();
+export const superSecret = config.require("supersecret");
+```
+
+{{% /choosable %}}
+
+{{% choosable language javascript %}}
+
+```javascript
+"use strict";
+const pulumi = require("@pulumi/pulumi");
+
+const config = new pulumi.Config();
+exports.superSecret = config.require("supersecret");
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+config = pulumi.Config()
+
+pulumi.export('superSecret',  config.require("supersecret"))
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		config := config.New(ctx, "")
+		superSecret := config.Require("supersecret")
+
+		ctx.Export("superSecret", pulumi.String(superSecret))
+		return nil
+	})
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+using Pulumi;
+
+class AnotherStack : Stack
+{
+    public AnotherStack()
+    {
+        var config = new Config();
+        this.SuperSecret = Output.Create(config.Require("supersecret"));
+    }
+
+    [Output("superSecret")] public Output<string> SuperSecret { get; set; }
+}
+```
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+Now we need to verify if the value is _actually_ encrypted. An easy way to do that is to try and export the secret value without access to the key. Let's unset the AWS_PROFILE environment variable and then rerun `Pulumi up`:
+
+```bash
+unset AWS_PROFILE
```

Comment:
Not sure that all users will be using an `AWS_PROFILE` to configure access.  Is there another way to highlight this that is more general?

Reply:
I've reworded this to make it more clear what `AWS_PROFILE` is and what other users would need to do.